cpp/include/raft/neighbors/detail/cagra/compute_distance_vpq.cuh

```diff
@@ -88,6 +88,26 @@ struct cagra_q_dataset_descriptor_t : public dataset_descriptor_base_t<half, DIS
   {
   }
 
+  template <uint32_t DATASET_BLOCK_DIM>
+  __device__ void copy_query(const DATA_T* const dmem_query_ptr,
+                             QUERY_T* const smem_query_ptr,
+                             const std::uint32_t query_smem_buffer_length)
+  {
+    for (unsigned i = threadIdx.x * 2; i < dim; i += blockDim.x * 2) {
+      half2 buf2 = {CUDART_ZERO_FP16, CUDART_ZERO_FP16};
+      if (i < dim) { buf2.x = static_cast<half>(static_cast<float>(dmem_query_ptr[i])); }
```

Comment:
Probably, we should replace the casts with `spatial::knn::detail::utils::mapping<float>{}` as well, right? In case if queries are 8-bit ints, the mapping will do normalization.